source/parser.hera

```diff
@@ -3864,44 +3864,93 @@ JSXTag
 # https://facebook.github.io/jsx/#prod-JSXElement
 JSXElement
   JSXSelfClosingElement
-  JSXOpeningElement JSXChildren? Whitespace? JSXClosingElement ->
-    // Check that tags match
-    if ($1[1] !== $4[2]) return $skip
-    return { type: "JSXElement", children: $0 }
-  # NOTE: Auto-closing element with indented children
-  JSXOpeningElement:open JSXMixedChildren:children InsertNewline InsertIndent ->
-    return {
-      type: "JSXElement",
-      children: children.jsxChildren.length ? [...$0, ["</", open[1], ">"]]
-                                            : [open.slice(0, -1), " />"]
+  # NOTE: In default JSX mode, JSX children must be properly indented,
+  # and closing tags are optional (but still allowed).
+  !CoffeeJSXEnabled PushJSXOpeningElement:open JSXMixedChildren?:children JSXOptionalClosingElement:close ->
+    module.JSXTagStack.pop()
```

Comment:
We might want to omit the assertion with `$0 = $0.slice(1)` above this line.

I think a later version of Hera will omit assertions from the results array but that's in the future.

Reply:
Oh, I assumed assertions were omitted. Thanks! I'll fix that in a bunch of places.